TypedSyntax/src/show.jl

```diff
@@ -113,12 +115,16 @@ function type_annotation_mode(node, @nospecialize(T); type_annotations::Bool, hi
     return type_annotate, pre, pre2, post
 end
 
-function show_annotation(io, @nospecialize(T), post, node, position; iswarn::Bool)
+function show_annotation(io, @nospecialize(T), post, node, position; iswarn::Bool, maxtypedepth)
     diagnostics = get(io, :diagnostics, nothing)
     inlay_hints = get(io, :inlay_hints, nothing)
 
     print(io, post)
     T_str = string(T)
+    if maxtypedepth !== nothing
+        sz = get(io, :displaysize, displaysize(io))::Tuple{Int, Int}
+        T_str = Base.type_depth_limit(T_str, max(sz[2], 120); maxdepth=maxtypedepth)
```

Comment:
As far as I can tell only max(sz[2], 120) or maxtypedepth is used not both. It seems better to control the string width directly so perhaps we should control that instead.

Reply:
I think this may still be useful when `maxdepthtype = nothing`?

Reply:
Yes you're right, sorry. But to try to explain myself more clearly, your issue was that some types that are printed are extremely long. It seems to me the second argument to `Base.type_depth_limit` controls the size of the printed type so is there a reason we don't set a limit for that instead of limiting the type depth?

Reply:
Yes, but for packages like JET/Cthulhu, it's sometimes helpful to set this depth explicitly. For example, in closures, the depth is too shallow and you cannot see types that you would like to.

Reply:
I think I'd prefer that users can toggle this, so that they can specify the level of type information.